src/storage/page_buffer.c

```diff
@@ -1967,37 +1972,59 @@ pgbuf_fix_release (THREAD_ENTRY * thread_p, const VPID * vpid, PAGE_FETCH_MODE f
 	   * page. */
 	}
     }
-  else if (fetch_mode == OLD_PAGE_IF_IN_BUFFER)
-    {
-      /* we don't need to fix page */
-      pthread_mutex_unlock (&hash_anchor->hash_mutex);
-      return NULL;
-    }
   else
     {
-      bufptr = pgbuf_claim_bcb_for_fix (thread_p, vpid, fetch_mode, hash_anchor, &perf, &retry);
-      if (bufptr == NULL)
+      /* The page is not found in the hash chain the caller is holding hash_anchor->hash_mutex */
+
+      if (er_errid () != NO_ERROR)
 	{
-	  if (retry)
-	    {
-	      retry = false;
-	      goto try_again;
-	    }
-	  ASSERT_ERROR ();
-	  return NULL;
+	  // error searching the hash chain
+	  pthread_mutex_unlock (&hash_anchor->hash_mutex);
+	  PGBUF_BCB_CHECK_MUTEX_LEAKS ();
+	  return nullptr;
 	}
-      buf_lock_acquired = true;
 
-#if defined(ENABLE_SYSTEMTAP)
-      if (fetch_mode == NEW_PAGE && pgbuf_hit == false)
+      if (fetch_mode == OLD_PAGE_IF_IN_BUFFER)
 	{
-	  pgbuf_hit = true;
+	  /* we don't need to fix page */
+	  pthread_mutex_unlock (&hash_anchor->hash_mutex);
+	  return NULL;
 	}
-      if (fetch_mode != NEW_PAGE)
+      else
```

Comment:
When there is a return in an if scope, else is not necessary. It adds two additional indentations and, if the else scope is quite big, it is difficult to follow the branches.

So sometimes a condition that returns:
```c
  if (cond)
    {
      ...
      return ...;
    }
  else
    {
      // else stuff
      ...
    }
```
it is preferable to rewrite as:
```c
  if (cond)
    {
      ...
      return ...;
    }

  // else stuff
  ...
```

I think this case qualifies for the second option.